marklogic-data-hub/src/test/java/com/marklogic/hub/flow/VerifyBatchTimestampTest.java

```diff
@@ -0,0 +1,43 @@
+package com.marklogic.hub.flow;
+
+import com.marklogic.client.io.Format;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.flow.FlowInputs;
+import com.marklogic.hub.flow.RunFlowResponse;
+import com.marklogic.hub.step.RunStepResponse;
+import com.marklogic.hub.test.Customer;
+import com.marklogic.hub.test.ReferenceModelProject;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class VerifyBatchTimestampTest extends AbstractHubCoreTest {
+
+    @Test
+    void test() {
+        installProjectInFolder("test-projects/custom-step-with-sleep");
+        
+        ReferenceModelProject project = new ReferenceModelProject(getHubClient());
+        doWithWriteBatcher(getHubClient().getStagingClient(), writeBatcher -> {
+        	writeBatcher.add(project.buildCustomerInstanceToWrite(new Customer(0, "Jane"), Format.JSON, null));
+        });
+        
+        RunFlowResponse response = runFlow(new FlowInputs("customFlowWithSleep"));
+        RunStepResponse stepResponse = response.getStepResponses().get("1");
+
+        assertEquals("completed step 1", stepResponse.getStatus());
+        assertEquals(1, stepResponse.getSuccessfulBatches());
+        
+        String start_time = stepResponse.getStepStartTime();
+        String end_time = stepResponse.getStepEndTime();
+        
+        // extraction out seconds value from timestamp, then round to nearest second
+        int t1 = (int) (Float.parseFloat(start_time.split(":")[2].split("-")[0]) + 0.5);
```

Comment:
Let's just do before/after with LocalDateTime classes; that's more reliable and more readable - i.e. looking at this, I have to do some thinking to figure out why either 5 or 55 is a correct value.

Reply:
Used your other suggestion